src/Avalonia.Visuals/Rendering/SceneGraph/VisualNode.cs

```diff
@@ -179,6 +179,11 @@ public void ReplaceDrawOperation(int index, IRef<IDrawOperation> operation)
         /// <param name="scene">The scene that the node is a part of.</param>
         public void SortChildren(Scene scene)
         {
+            if (_children == null || _children.Count <= 1)
+            {
+                return;
+            }
+
             var keys = new List<long>();
```

Comment:
I know it is not really part of this pr but you know the final length of keys so it might save cpu to set in the constructor

Reply:
Good point! Might as well make that change here.